test/tests_multiple_instances.js

```diff
@@ -0,0 +1,126 @@
+        test.assertEquals(message[0], 'Processing queued call');
+        var params = JSON.parse(message[1]);
+        test.assertEquals(params[0], "YOUR_APP_KEY3");
+        test.assertEquals(params[1], "add_event");
+        test.assertEquals(params[2].key, "third_app");
+    });
+    tests.push(function (message){
+        test.assertEquals(message[0], '[YOUR_APP_KEY3] Adding event: ');
+        var params = JSON.parse(message[1]);
+        test.assertEquals(params.key, "third_app");
+        test.assertEquals(params.count, 1);
+    });
+    tests.push(function (message){
+        test.assertEquals(message[0], 'Processing request');
+        var params = JSON.parse(message[1]);
+
+        test.assertEquals(params.app_key, "YOUR_APP_KEY1");
+        test.assert(exists(params.device_id));
+        test.assert(exists(params.timestamp));
+        test.assert(exists(params.hour));
+        test.assert(exists(params.dow));
+        
+        params.events = JSON.parse(params.events);
+        test.assertEquals(params.events.length, 1);
+        
+        test.assertEquals(params.events[0].key, 'first_app');
+        test.assertEquals(params.events[0].count, 1);
+    });
+    tests.push(function (message){
+        test.assertEquals(message[0], 'Sending XML HTTP request');
+    });
+    tests.push(function (message){
+        test.assertEquals(message[0], '[YOUR_APP_KEY2] Processing request');
+        var params = JSON.parse(message[1]);
+
+        test.assertEquals(params.app_key, "YOUR_APP_KEY2");
+        test.assert(exists(params.device_id));
+        test.assert(exists(params.timestamp));
+        test.assert(exists(params.hour));
+        test.assert(exists(params.dow));
+        
+        params.events = JSON.parse(params.events);
+        test.assertEquals(params.events.length, 1);
+        
+        test.assertEquals(params.events[0].key, 'second_app');
+        test.assertEquals(params.events[0].count, 1);
+    });
+    tests.push(function (message){
+        test.assertEquals(message[0], '[YOUR_APP_KEY2] Sending XML HTTP request');
+    });
+    tests.push(function (message){
+        test.assertEquals(message[0], '[YOUR_APP_KEY3] Processing request');
+        var params = JSON.parse(message[1]);
+
+        test.assertEquals(params.app_key, "YOUR_APP_KEY3");
+        test.assert(exists(params.device_id));
+        test.assert(exists(params.timestamp));
+        test.assert(exists(params.hour));
+        test.assert(exists(params.dow));
+        
+        params.events = JSON.parse(params.events);
+        test.assertEquals(params.events.length, 1);
+        
+        test.assertEquals(params.events[0].key, 'third_app');
+        test.assertEquals(params.events[0].count, 1);
+    });
+    tests.push(function (message){
+        test.assertEquals(message[0], '[YOUR_APP_KEY3] Sending XML HTTP request');
+    });
+    casper.removeAllListeners('remote.message');
+    casper.on('remote.message', function(message) {
```

Comment:
Codacy found an issue: ['casper' is not defined.](https://app.codacy.com/manual/ar2rsawseen/countly-sdk-web/pullRequest?prid=6356797)